### PR TITLE
openssl: retrieve reported libressl version at runtime

### PR DESCRIPTION
**Pull request by jay**
LibreSSL added a OpenSSL_version_num that always returns 0x020000000L.
That conflicted with the workaround function we were using to return the
compile-time LibreSSL actual version. This change removes that
workaround in favor of extracting the actual version at runtime.

Fixes https://github.com/curl/curl/issues/2319

### CODE REVIEW

File: lib/vtls/openssl.c [hunk 1]
```diff
@@ -3519,8 +3512,23 @@ static ssize_t ossl_recv(struct connectdata *conn, /* connection data */
 
 static size_t Curl_ossl_version(char *buffer, size_t size)
 {
+  int rc = -1;
 #ifdef OPENSSL_IS_BORINGSSL
-  return snprintf(buffer, size, OSSL_PACKAGE);
+  /* no version extraction method known */
+#elif LIBRESSL_VERSION_NUMBER
```
Comment (bagder) [inline lib/vtls/openssl.c]: #elif defined(LIBRESSL_VERSION_NUMBER)

Comment (jay) [reply lib/vtls/openssl.c]: Thanks, fixed

**Comment by bagder** [comment]
Note that this causes a CI error on the libressl build on macOS

**Comment by bagder** [comment]
Rebase and try again?

**Comment by bagder** [comment]
Is this something you still intend to work on?

**Comment by bagder** [comment]
The work on this has clearly stopped so let's close this for now and we can reopen if the work resumes!

**Pull request closed without merging**